backend/kale/common/snapshotutils.py

```diff
@@ -0,0 +1,119 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  To allow a notebook to create Rook CephFS snapshot a ClusterRole
+#  and RoleBinding for the default-editor service account in the
+#  given namespace must be applied. Below are examples for use with the
+#  namespace "admin".
+#
+#  !!!WARNING!!!
+#  Might not be secure, only use for testing
+#
+#  apiVersion: rbac.authorization.k8s.io/v1
+#  kind: ClusterRole
+#  metadata:
+#    name: snapshot-access
+#  rules:
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshots"]
+#      verbs: ["create", "get", "list", "watch", "patch", "delete"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotcontents"]
+#      verbs: ["create", "get", "list", "watch", "update", "delete"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotclasses"]
+#      verbs: ["get", "list", "watch"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotcontents/status"]
+#      verbs: ["update"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshots/status"]
+#      verbs: ["update"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotclasses"]
+#      verbs: ["get", "list", "watch"]
+#    - apiGroups: ["storage.k8s.io"]
+#      resources: ["storageclasses"]
+#      verbs: ["get", "list"]
+#
+#  apiVersion: rbac.authorization.k8s.io/v1
+#  kind: RoleBinding
+#  metadata:
+#    name: allow-snapshot-nb-admin
+#    namespace: admin
+#  subjects:
+#  - kind: ServiceAccount
+#    name: default-editor
+#    namespace: admin
+#  roleRef:
+#    kind: ClusterRole
+#    name: snapshot-access
+#    apiGroup: rbac.authorization.k8s.io
+
+import logging
+
+from kale.common import podutils, k8sutils
+
+NOTEBOOK_SNAPSHOT_COMMIT_MESSAGE = """\
+This is a snapshot of notebook {} in namespace {}.
+This snapshot was created by Kale in order to clone the volumes of the notebook
+and use them to spawn a Kubeflow pipeline.\
+"""
+
+log = logging.getLogger(__name__)
+
+
+def snapshot_pvc(snapshot_name, pvc_name, image="", path="", **kwargs):
+    """Perform a snapshot over a PVC."""
+    snapshot_resource = {
+        "apiVersion": "snapshot.storage.k8s.io/v1beta1",
+        "kind": "VolumeSnapshot",
+        "metadata": {
+            "name": snapshot_name,
+            "annotations": {
+                "container_image": image,
+                "volume_path": path
+            },
+            "labels": kwargs
+        },
+        "spec": {
+            "volumeSnapshotClassName": get_snapshotclass_name(pvc_name),
+            "source": {"persistentVolumeClaimName": pvc_name}
+        }
+    }
+    co_client = k8sutils.get_co_client()
+    namespace = podutils.get_namespace()
+    log.info("Taking a snapshot of PVC %s in namespace %s ..."
+             % (pvc_name, namespace))
```

Comment:
You can use commas instead of `%` when logging. So this becomes

```python
log.info("Taking a snapshot of PVC %s in namespace %s ...", (pvc_name, namespace))
```